components/plan-build/plan.sh

```diff
@@ -35,6 +35,7 @@ do_build() {
   sed \
     -e "s,#!/bin/bash$,#!$(pkg_path_for bash)/bin/bash," \
     -e "s,^HAB_PLAN_BUILD=.*$,HAB_PLAN_BUILD=$pkg_version/$pkg_release," \
+    -e "s,^pkg_target=.*$,pkg_target='$pkg_target'," \
```

Comment:
Is this only for the case where we want to replace `pkg_target='@@pkg_target@@'`? If so, I'd suggest making that explicit:
```
    -e "s,^pkg_target='@@pkg_target@@'\$,pkg_target='$pkg_target',"
```
since that makes this code much easier to discover. And if it's not, I'd still suggest escaping the `$` in the regex to make it clear that it's behaving as the end-of-line anchor and not a [special shell parameter](https://www.gnu.org/software/bash/manual/html_node/Special-Parameters.html#Special-Parameters).

Reply:
I waffled on this question as well, but in the interests of local context (i.e. the approach in the previous 2 lines), I went with this one.

So, in order to get the best of both worlds,  I addressed this and updated the previous 2 lines for better internal consistency.

Good catch!